pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -166,9 +166,7 @@ func (d *defaultDriver) Reconcile() *reconciler.Results {
 		))
 
 	// always update the elasticsearch state bits
-	if observedState.ClusterInfo != nil && observedState.ClusterHealth != nil {
-		d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState)
-	}
+	d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState)
```

Comment:
Can we remove the same call on L246 then?

Reply:
Good question. I'm looking into it.

Reply:
I think we have a spaghetti plate to handle here. We seem to update the ES status at multiple places (sometimes deep down in nodes reconciliation), multiple times.
I think it's better to extract it to its own issue & PR.

Reply:
Works for me.

Reply:
https://github.com/elastic/cloud-on-k8s/issues/2403